Dockerfile

```diff
@@ -49,20 +51,25 @@ RUN --mount=type=bind,target=.,rw <<EOT
   fi
 EOT
 
-FROM golangci/golangci-lint:${GOLANGCI_LINT_VERSION}-alpine AS golangci-lint
+FROM golangci/golangci-lint:${GOLANGCI_LINT_VERSION} AS golangci-lint
 FROM gobase AS lint
-RUN apk add musl-dev gcc libsecret-dev pass
+ARG DEBIAN_FRONTEND
+RUN apt-get install -y binutils gcc libc6-dev libgcc-10-dev libsecret-1-dev pkg-config
 RUN --mount=type=bind,target=. \
     --mount=type=cache,target=/root/.cache \
     --mount=from=golangci-lint,source=/usr/bin/golangci-lint,target=/usr/bin/golangci-lint \
     golangci-lint run ./...
 
 FROM gobase AS base
 ARG TARGETPLATFORM
-RUN xx-apk add musl-dev gcc libsecret-dev pass
+ARG DEBIAN_FRONTEND
+RUN xx-apt-get install -y binutils gcc libc6-dev libgcc-10-dev libsecret-1-dev pkg-config
 
-FROM base AS test
-RUN xx-apk add gnome-keyring gpg-agent gnupg-gpgconf pass
+FROM golang:${GO_VERSION}-alpine AS test
```

Comment:
unfortunately can't make sandboxed test work with debian base image :disappointed:

it seems `gpg` or `pass` does not behave like one alpine:

```
#15 0.721 ok    github.com/docker/docker-credential-helpers/credentials 0.002s  coverage: 58.5% of statements
#15 0.752 === RUN   TestPassHelper
#15 0.752     pass_test.go:20: exit status 1: gpg: WARNING: unsafe permissions on homedir '/root/.gnupg'
#15 0.752         gpg: 0xD523BD50DD70B0BA: There is no assurance this key belongs to the named user
#15 0.752         gpg: cannot open '/dev/tty': No such device or address
#15 0.752         Password encryption aborted.
#15 0.752
#15 0.752 --- FAIL: TestPassHelper (0.02s)
```

Reply:
ah found the issue, on debian based we have to change the trust level command to:

```
echo '5\ny\n' | gpg --command-fd 0 --no-tty --edit-key 7D851EB72D73BDA0 trust
```

Reply:
For the permissions, we could probably also try to fix by setting the directory permissions when we create the directory (I recall I made that change when I was looking into things)

Reply:
Yeah, was thinking; add a `mkdir` (and `chmod` if needed) here; https://github.com/docker/docker-credential-helpers/blob/2f246b8570b19c8ddb629149c7590a58e3715f06/Dockerfile#L70